### ISSUE TITLE
WebInternalFrame issues

### ISSUE THREAD

**Issue by mgarin** [mgarin/weblaf#27]
There are a few problems with internal frames:
- Title cannot be set using `setTitle()` method
- Title buttons are focusable
- Its almost impossible to resize frames by dragging their sides
- There are some minor painting issues

**Original issues:**
http://weblookandfeel.com/forum/viewtopic.php?f=4&t=139
http://weblookandfeel.com/forum/viewtopic.php?f=4&t=56

**Comment by Abu-Abdullah** [comment]
Also buttons are not rendered correctly for setClosable/setMaximizable ... the one in the demo is rendering correctly

**Comment by mgarin** [comment]
Don't worry, internal frame fixes are coming out pretty soon :)
I will post an additional update here when it is out and can be checked out.

**Comment by renato** [comment]
@ndandoulakis thanks a lot!

I'm using setTitle and it's working with this hack:
```
setTitle("Anything");
getDesktopIcon().updateUI();
updateUI();
```

**Comment by iMichka** [comment]
Thanks for the setTitle hint. ```updateUI();``` seems enough to fix it. Maybe overkill to call this but works as a temporary workaround.

**Comment by iMichka** [comment]
The hack to use setBorder does no more work (not sure if this is a change in Weblaf or due to some value changing/resetting on my side). It seems to me that setBorder does nothing to the frame (the border size can not be set from outside of WebLaf).

One solution I found is to change the border from 1,1,1,1 instead of 0,0,0,0 in the code here:
https://github.com/mgarin/weblaf/blob/master/modules/ui/src/com/alee/laf/desktoppane/WebInternalFrameUI.java#L99

Not sure this is the definitive solution to this problem.

**Comment by didingtech** [comment]
@mgarin hi,did you fixed it?

**Comment by didingtech** [comment]
@mgarin how can i changge the WebInternalFrame L&F?

**Comment by mgarin** [comment]
> @mgarin hi,did you fixed it?

Not yet. Fixes for `WebInternalFrame` will be available with the final release of v1.29. I am still working on bringing a huge chunk of changes from `styling` branch into `master` including internal frame ones.

> @mgarin how can i changge the WebInternalFrame L&F?

You can provide different UI class canonical name into `WebLookAndFeel.internalFrameUI` field.
Though it is recommended to have a UI class based on `WebInternalFrameUI` to avoid issues.

**Comment by didingtech** [comment]
@mgarin  thanks a lot! :)

**Comment by didingtech** [comment]
@mgarin how can i changge the WebInternalFrame font? I'm using  "WebLookAndFeel.internalFrameFont" ,but it doesn't work

**Comment by iMichka** [comment]
Another thing I just realized, and probably related: if SetClosable() is called on a frame, it will not be updated. calling updateUI() is also needed here. Works with the workaround so not so urgent.

**Comment by mgarin** [comment]
@mark-co Sorry for late answer - for some reason I didn't get email update on this issue - in current version there is no convenient way to change it. There will be in the final release. As a workaround - you can access the title label through rootpane component children and modify font manually there.

@iMichka Thanks for the update. Fixes for internal frame will be available in the final v1.29 release and will probably be included in new preview build next week.
